api/dbadapters/index.ts

```diff
@@ -71,8 +72,10 @@ register("redshift", RedshiftDbAdapter);
 register("snowflake", SnowflakeDbAdapter);
 register("sqldatawarehouse", SQLDataWarehouseDBAdapter);
 
+export type QueryOrAction = string | dataform.Table | dataform.Operation | dataform.Assertion;
```

Comment:
nit: IMO use IQueryOrAction for types as well as interfaces, it makes it clear this is just a type, not a class

Reply:
I don't like that because you can't do some things with a type that you can do with an interface. Surely if we prefer using the `dataform.Table` protos now, that doesn't start with an `I` and acts as a type